Add Spot Listing UI, Marketplace View, and Backend Integration

nathanchamorro1 opened the pull request:
This PR adds the complete spot listing creation UI and the ability to fetch and display active listings from the backend. It includes a form where users can select the parking structure, floor, time, and credits, and it integrates the Listing model with the GET /get_listings endpoint.

This update also introduces the initial marketplace view, allowing users to see active listings along with their price and location.

For now, vehicle selection is handled through a temporary short form that lets users enter basic vehicle information. This is only a placeholder and will be removed once #53 is merged in, since that PR contains the improved vehicle management screen.

The backend is currently using a dummy account for listing creation. It is not yet connected to the authentication system. This is because I wanted to focus on UI and listing functionality. Hooking it up to the auth UUID is straightforward.

nathanchamorro1 committed:
implented listing and buy screen, integrate vehicle creation, map listings from backend, and update marketplace rendering

nathanchamorro1 committed:
Fixed CI...

nathanchamorro1 committed:
Merge branch 'main' into 42-spot-listings

nathanchamorro1 committed:
got rid of warning for deprecated value

santi224m commented:
Looks good. For ```lib/services/titanpark_api.dart```, can you use the ```config.dart``` file that Sharon added for the base URL? I have mine configured to use the ```parking.titanpark.online``` server API, but when I run your branch I get the following error since it uses ```final String baseUrl = 'http://127.0.0.1:8000';```. This is on the "Buy Parking Spot" page.

<img width="409" height="908" alt="image" src="https://github.com/user-attachments/assets/e6acf594-01ad-451c-972e-76938546124d" />

nathanchamorro1 committed:
fixes to issues + local baseurl fix

nathanchamorro1 commented:
I updated TitanParkApi to use kParkingApiBaseUrl from config.dart instead of the localhost URL, so it now correctly points to parking.titanpark.online by default.

I also fixed the other two issues related to listing parsing and structure mapping.

santi224m commented:
I'm still getting the same error and it looks like ```lib/services/titanpark_api.dart``` still manually sets the base API url to ```http://127.0.0.1:8000```. Maybe you forgot to add the changes to your commit?
